### PR TITLE
asyncOpenSelectedProjects will return project that correspond to the candidates

### PR DESCRIPTION
**Pull request by sdedic**
From an offline report:

Open a Maven GCN project in VSCode with NBLS installed (no RH Java). Go to 'Projects' view an try to invoke 'Compile Project' action on the top level project's node. The following exception is thrown

```
WARNING [org.netbeans.modules.java.lsp.server.protocol.Server]: Error occurred during LSP message dispatch
java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:359)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at org.netbeans.modules.java.lsp.server.protocol.WorkspaceServiceImpl.lambda$executeCommand$3(WorkspaceServiceImpl.java:226)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
Caused: java.util.concurrent.CompletionException
```

This exception happes because if a multiproject root directory (or a file with the root project itself, not a subproject) is passed, NBLS opens all contained projects - and returns the whole subtree. The original code attempted to match the projects against the input fileobjects and failed since there were more projects opened than the input fileobjects. 

The original intention of `asyncOpenSelectedProjects` was to return projects that correspond to the candidates after their open / prime. This PR changes the behaviour so that `asyncOpenSeelectedProjects` will return Project objects that correspond to individual input FileObjects, in the same order. If a FileObject is not part of any project, its corresponding array slot will be `null`.  I adjusted the callers to check for `null` values, hopefully.

### CODE REVIEW

**Review by MartinBalin — APPROVED** [review]

**Comment by sdedic** [comment]
OK, bad guy on initialization identified. Now I revert all the debugging crap.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2616,6 +2616,8 @@ jobs:
     needs: base-build
     runs-on: ubuntu-latest
     timeout-minutes: 60
+    env:
+      netbeans.extra.options: -J-Dnetbeans.logger.console=true
```
Comment (dbalek) [inline .github/workflows/main.yml:2620]: Debugging leftovers?

**Review by dbalek — APPROVED** [review]
Looks fine to me

File: .github/workflows/main.yml (hunk 1 above)
Comment (sdedic) [reply .github/workflows/main.yml]: Removed from commit d25a4c2

**Pull request merged**